extensions/standard-processors/processors/RouteText.cpp

```diff
@@ -0,0 +1,488 @@
+                      "Two segments that have the same Group but different Relationships will never be placed into the same FlowFile.")
+    ->build());
+
+const core::Property RouteText::GroupingFallbackValue(
+    core::PropertyBuilder::createProperty("Grouping Fallback Value")
+    ->withDescription("If the 'Grouping Regular Expression' is specified and the matching fails, this value will be considered the group of the segment.")
+    ->withDefaultValue<std::string>("")
+    ->build());
+
+const core::Property RouteText::SegmentationStrategy(
+    core::PropertyBuilder::createProperty("Segmentation Strategy")
+    ->withDescription("Specifies what portions of the FlowFile content constitutes a single segment to be processed.")
+    ->isRequired(true)
+    ->withDefaultValue<std::string>(toString(Segmentation::PER_LINE))
+    ->withAllowableValues<std::string>(Segmentation::values())
+    ->build());
+
+const core::Relationship RouteText::Original("original", "The original input file will be routed to this destination");
+
+const core::Relationship RouteText::Unmatched("unmatched", "Segments that do not satisfy the required user-defined rules will be routed to this Relationship");
+
+const core::Relationship RouteText::Matched("matched", "Segments that satisfy the required user-defined rules will be routed to this Relationship");
+
+RouteText::RouteText(const std::string& name, const utils::Identifier& uuid)
+    : core::Processor(name, uuid), logger_(logging::LoggerFactory<RouteText>::getLogger()) {}
+
+void RouteText::initialize() {
+  setSupportedProperties({
+     RoutingStrategy,
+     MatchingStrategy,
+     TrimWhitespace,
+     IgnoreCase,
+     GroupingRegex,
+     GroupingFallbackValue,
+     SegmentationStrategy
+  });
+  setSupportedRelationships({Original, Unmatched, Matched});
+}
+
+static std::regex to_regex(const std::string& str) {
+  return std::regex(str);
+}
+
+void RouteText::onSchedule(core::ProcessContext* context, core::ProcessSessionFactory* /*sessionFactory*/) {
+  routing_ = utils::parseEnumProperty<Routing>(*context, RoutingStrategy);
+  matching_ = utils::parseEnumProperty<Matching>(*context, MatchingStrategy);
+  context->getProperty(TrimWhitespace.getName(), trim_);
+  case_policy_ = context->getProperty<bool>(IgnoreCase).value_or(false) ? CasePolicy::IGNORE_CASE : CasePolicy::CASE_SENSITIVE;
+  group_regex_ = context->getProperty(GroupingRegex) | utils::map(to_regex);
+  segmentation_ = utils::parseEnumProperty<Segmentation>(*context, SegmentationStrategy);
+  context->getProperty(GroupingFallbackValue.getName(), group_fallback_);
+}
+
+class RouteText::ReadCallback : public InputStreamCallback {
+  using Fn = std::function<void(Segment)>;
+
+ public:
+  explicit ReadCallback(Segmentation segmentation, Fn&& fn) : segmentation_(segmentation), fn_(std::move(fn)) {}
```

Comment:
two-parameter constructors (also `OwningSearcher`) don't need to be explicit

Reply:
removed `explicit`